packages/manager/modules/nasha/src/dashboard/dashboard.routing.js

```diff
@@ -60,8 +58,20 @@ export default /* @ngInject */ ($stateProvider) => {
       },
       nashaApiUrl: /* @ngInject */ (baseApiUrl, serviceName) =>
         `${baseApiUrl}/${serviceName}`,
-      numberOfPartitions: /* @ngInject */ (nashaApiUrl, $http) =>
-        $http.get(`${nashaApiUrl}/partition`).then(({ data }) => data.length),
+      partitionAllocatedSize: /* @ngInject */ (iceberg, nashaApiUrl) =>
+        iceberg(`${nashaApiUrl}/partition`)
+          .query()
+          .expand('CachedObjectList-Pages')
+          .execute()
+          .$promise.then(({ data }) => {
+            let partitionAllocatedSize = 0;
+            data.forEach((partition) => {
+              partitionAllocatedSize += partition.size;
+            });
+            return partitionAllocatedSize;
```

Comment:
**niptick**
```suggestion
            return data.reduce((totalSize, partition) => {
              return totalSize += partition.size
            }, 0);
```

You can also replace the return (in suggestion), by shortcuts

Reply:
With the suggestion I have the message: "Return statement should not contain assignment.eslint[no-return-assign](https://eslint.org/docs/rules/no-return-assign)"

Or you have to:

```
.$promise.then(({ data }) => {
            return data.reduce((totalSize, partition) => {
              const total = totalSize + partition.size;
              return total;
            }, 0);
          }),
          ```

Reply:
As this it's working:

```
   .$promise.then(({ data }) =>
            data.reduce(
              (totalSize, partition) => totalSize + partition.size,
              0,
            ),
          ),
```

Reply:
It's applied